blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyActivity.java

```diff
@@ -747,11 +745,27 @@ public void onClick(View v) {
      * @see #getCodeGenerationCallback()
      */
     protected void onRunCode() {
-        mCodeGeneratorManager.requestCodeGeneration(
-            getBlockDefinitionsJsonPaths(),
-            getGeneratorsJsPaths(),
-            mWorkspaceFragment.getWorkspace(),
-            getCodeGenerationCallback());
+        Workspace workspace = mWorkspaceFragment.getWorkspace();
+        if (workspace.hasBlocks()) {
+            try {
+                final StringOutputStream serialized = new StringOutputStream();
+                workspace.serializeToXml(serialized);
+                CodeGenerationRequest codeGenerationRequest =
+                    new CodeGenerationRequest(
+                        serialized.toString(),
+                        getCodeGenerationCallback(),
+                        getBlockDefinitionsJsonPaths(),
+                        getGeneratorsJsPaths());
+                mCodeGeneratorManager.requestCodeGeneration(codeGenerationRequest);
+            } catch (BlocklySerializerException e) {
+                Log.wtf(TAG, e);
+                Toast.makeText(this, "Code generation failed.",
+                    Toast.LENGTH_LONG).show();
+
+            }
+        } else {
+            Log.i(TAG, "No blocks in workspace. Skipping run request.");
+        }
```

Comment:
Because this is going to have merge conflicts with some upcoming changes, I'd appreciate if you can update the PR to target the `toolbox-refactor` branch, and give me write access to this branch.

Thanks.